### PR TITLE
feature: add Azure Virtual Network Gateway scanner

### PR DESCRIPTION
**Pull request by vanwinkelseppe**
# Description

Add a scanner for Azure Virtual Network Gateway

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #206

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Run linter
* [x] Unit tests passing

### CODE REVIEW

**Comment by vanwinkelseppe** [comment]
@cmendible 
I'm not 100% familiar with the Go SDK for Azure, but do you have an idea where we find the Azure VPN Gateway SKU & Generation in the properties? I wasn't able to locate them.

You have any requests for other scan rules?

Thanks!

File: internal/scanners/vpng/rules.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			Url: "https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-resources?tabs=json",
+		},
+		"vpng-004": {
+			Id:             "vpng-004",
+			Category:       scanners.RulesCategoryHighAvailability,
+			Recommendation: "VPN Gateway should have a SLA",
+			Impact:         scanners.ImpactHigh,
+			Eval: func(target interface{}, scanContext *scanners.ScanContext) (bool, string) {
+				return false, "99.9%"
+				//TODO: Filter SKU based on tier (BASIC / Or others)
```
Comment (cmendible) [inline internal/scanners/vpng/rules.go:58]: Let's do this before merging

**Comment by cmendible** [comment]
@vanwinkelseppe I'm about to merge: https://github.com/Azure/azqr/pull/214 so please check if the new SDK for networking helps you with the SLA rule we are reviewing

**Comment by vanwinkelseppe** [comment]
@cmendible will do thanks!

**Comment by vanwinkelseppe** [comment]
@cmendible Still no SKU on the VPNGateway, bypassed it by querying for the VirtualNetworkGateway.

Ready for review, have to do the list of rules, but with all the merges I'll do it when it looks ready to go!

File: docs/content/en/docs/Recommendations/_index.md
Comment (cmendible) [inline docs/content/en/docs/Recommendations/_index.md:1]: please rebase. There is no need to manually update this file. It's generated with each build

**Review by cmendible — CHANGES_REQUESTED** [review]

**Review by cmendible — CHANGES_REQUESTED** [review]
Hey! Check the following branch: https://github.com/Azure/azqr/tree/vpn I made some changes so we only query for Virtual Network Gateway objects  and take decisions based on the type.

If you are ok with it  merge it into your PR and if possible add an Availability Zone rule based on the service tier.

**Comment by vanwinkelseppe** [comment]
> Hey! Check the following branch: https://github.com/Azure/azqr/tree/vpn I made some changes so we only query for Virtual Network Gateway objects and take decisions based on the type.
> 
> If you are ok with it merge it into your PR and if possible add an Availability Zone rule based on the service tier.

Will do tonight or tomorrow! Thanks for the refactor!

**Comment by vanwinkelseppe** [comment]
@cmendible, uncertain for the availability zones.
I've checked the properties on virtualnetworkgateway, but couldn't find anything related to it except ExtendedLocation. But uncertain what this actually means. I've queried some resources on my environment via api to see, but it seems to be never filled in.

I did make a list of which gateway skus can enable them:
[Express](https://learn.microsoft.com/en-us/azure/expressroute/expressroute-about-virtual-network-gateways):
- ErGw1AZ
- ErGw2AZ
- ErGw3AZ

[VPN](url)
- Gen 1 - VpnGw1AZ
- Gen 1 - VpnGw2AZ
- Gen 1 - VpnGw3AZ
- Gen 2 - VpnGw2AZ
- Gen 2 - VpnGw3AZ
- Gen 2 - VpnGw4AZ
- Gen 2 - VpnGw5AZ

File: docs/content/en/docs/Overview/_index.md [hunk 2]
```diff
@@ -102,6 +102,7 @@ To learn more about the recommendations used by **Azure Quick Review (azqr)**, y
 * Azure Virtual Machine Scale Set
 * Azure Virtual Network
 * Azure Virtual WAN
+* Azure VPN Gateway
```
Comment (cmendible) [inline docs/content/en/docs/Overview/_index.md:105]: Let's add the other 2 types: local and express route

**Review by cmendible — CHANGES_REQUESTED** [review]

**Review by cmendible — APPROVED** [review]
LGTM

File: internal/scan.go [hunk 3]
```diff
@@ -7,6 +7,7 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"github.com/Azure/azqr/internal/scanners/vgw"
```
Comment (cmendible) [inline internal/scan.go:10]: Can we move this line to the bottom as the rest of imports?

**Review by cmendible — CHANGES_REQUESTED** [review]

**Review by cmendible — DISMISSED** [review]

**Comment by cmendible** [comment]
> @cmendible, uncertain for the availability zones. I've checked the properties on virtualnetworkgateway, but couldn't find anything related to it except ExtendedLocation. But uncertain what this actually means. I've queried some resources on my environment via api to see, but it seems to be never filled in.
> 
> I did make a list of which gateway skus can enable them: [Express](https://learn.microsoft.com/en-us/azure/expressroute/expressroute-about-virtual-network-gateways):
> 
> * ErGw1AZ
> * ErGw2AZ
> * ErGw3AZ
> 
> [VPN](url)
> 
> * Gen 1 - VpnGw1AZ
> * Gen 1 - VpnGw2AZ
> * Gen 1 - VpnGw3AZ
> * Gen 2 - VpnGw2AZ
> * Gen 2 - VpnGw3AZ
> * Gen 2 - VpnGw4AZ
> * Gen 2 - VpnGw5AZ

@vanwinkelseppe let's use the SKU, and if SKU (to lower) contains `az` then the Gateway is compliant.

**Comment by cmendible** [comment]
Closed in favor of: https://github.com/Azure/azqr/pull/237

**Pull request closed without merging**